pkg/webhook/admission/webhook_test.go

```diff
@@ -0,0 +1,379 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/mattbaird/jsonpatch"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/types"
+)
+
+var _ = Describe("admission webhook", func() {
+	Describe("validating webhook", func() {
+		var alwaysAllow, alwaysDeny *fakeHandler
+		var req *http.Request
+		var w *fakeResponseWriter
+		var wh *Webhook
+		BeforeEach(func(done Done) {
+			alwaysAllow = &fakeHandler{
+				fn: func(ctx context.Context, req Request) Response {
+					return Response{
+						Response: &admissionv1beta1.AdmissionResponse{
+							Allowed: true,
+						},
+					}
+				},
+			}
+			alwaysDeny = &fakeHandler{
+				fn: func(ctx context.Context, req Request) Response {
+					return Response{
+						Response: &admissionv1beta1.AdmissionResponse{
+							Allowed: false,
+						},
+					}
+				},
+			}
+			req = &http.Request{
+				Header: http.Header{"Content-Type": []string{"application/json"}},
+				Body:   nopCloser{Reader: bytes.NewBufferString(`{"request":{}}`)},
+			}
+
+			w = &fakeResponseWriter{}
+			close(done)
+		})
+
+		Context("multiple handlers can be invoked", func() {
+			BeforeEach(func(done Done) {
+				wh = &Webhook{
+					Type:     types.WebhookTypeValidating,
+					Handlers: []Handler{alwaysAllow, alwaysDeny},
+					KVMap:    map[string]interface{}{"foo": "bar"},
+				}
+				close(done)
+			})
+
+			It("should deny the request", func() {
+				expected := []byte(`{"response":{"uid":"","allowed":false}}
+`)
+				wh.ServeHTTP(w, req)
+				Expect(w.response).NotTo(BeNil())
+				Expect(w.response).To(Equal(expected))
+				Expect(alwaysAllow.invoked).To(BeTrue())
+				Expect(alwaysDeny.invoked).To(BeTrue())
+				Expect(alwaysAllow.valueFromContext).To(Equal("bar"))
+				Expect(alwaysDeny.valueFromContext).To(Equal("bar"))
+			})
+		})
+
+		Context("validating webhook should return if one of the handler denies", func() {
+			BeforeEach(func(done Done) {
+				wh = &Webhook{
+					Type:     types.WebhookTypeValidating,
+					Handlers: []Handler{alwaysDeny, alwaysAllow},
+					KVMap:    map[string]interface{}{"foo": "bar"},
+				}
+				close(done)
+			})
+
+			It("should deny the request", func() {
+				expected := []byte(`{"response":{"uid":"","allowed":false}}
+`)
+				wh.ServeHTTP(w, req)
+				Expect(w.response).NotTo(BeNil())
+				Expect(w.response).To(Equal(expected))
+				Expect(alwaysDeny.invoked).To(BeTrue())
+				Expect(alwaysDeny.valueFromContext).To(Equal("bar"))
+				Expect(alwaysAllow.invoked).To(BeFalse())
```

Comment:
So the handlers are executed with respect to their order in the `Handlers` slice. Is that something users need to be aware of? Do you need to add some doc for that?